scripts/globals/job_utils/beastmaster.lua

```diff
@@ -0,0 +1,132 @@
+-----------------------------------
+-- beastmaster Job Utilities
+-----------------------------------
+
+-----------------------------------
+xi = xi or {}
+xi.job_utils = xi.job_utils or {}
+xi.job_utils.beastmaster = xi.job_utils.beastmaster or {}
+
+local function getCharmDuration(charmer, target)
+    local charmDuration = 0
+
+    -- Calculate base duration and dLvl
+    local baseCharmDuration = math.floor(1.25 * charmer:getStat(xi.mod.CHR) + 150)
+    local dLvl = charmer:getMainLvl() - target:getMainLvl()
+
+    -- Default multiplier for dLvl -6 or lower
+    local dLvlCharmMult = 1 / 24
+
+    if dLvl >= -6 and dLvl < 9 then
+        -- Quintic fitting of duration multiplier as function of dLvl (r^2 > 0.999)
```

Comment:
Very fancy, where did you get the information for this? Might be worth linking to a site that shows a visual graphing of this so it can be easily compared to what's in the wikis etc.

Reply:
![dlvl_to_charm_duration_func_fitting](https://github.com/LandSandBoat/server/assets/156478534/144e95e3-7e92-4fee-b135-c4a2a46ccdf8)

Yeah, this is the fitting of the function to the data points from the testing, including fitting metrics.

Reply:
You've generated this, how about a link to the site where you did this so people can look it up themselves? Once the context of this PR is gone these are just numberwang

Reply:
> You've generated this, how about a link to the site where you did this so people can look it up themselves? Once the context of this PR is gone these are just numberwang

Yeah the data points are from the table [here](https://www.bg-wiki.com/ffxi/Charm_Duration) which was validated by retail testing by Monti on 11/2022. The fitting was done by quintic least squares fitting and can be seen [here](https://mycurvefit.com/index.html?action=openshare&id=358a5d99-4499-4a6a-bbfe-0a667739335c) via mycurvefit. Do you want me to put the links in the code itself? Thanks for the comments so far!

Reply:
Yeah, a comment in the code please. Like I say, once the context of the PR is gone, there's nothing else to go on

Reply:
> Yeah, a comment in the code please. Like I say, once the context of the PR is gone, there's nothing else to go on



> Yeah, a comment in the code please. Like I say, once the context of the PR is gone, there's nothing else to go on

Yep, ok, I added comments.